accepted/0000-uyuni-microos-management.md

```diff
@@ -0,0 +1,142 @@
+- Feature Name: (fill with a unique ident, my_awesome_feature)
+- Start Date: (fill with today's date, YYYY-MM-DD)
+
+# Summary
+[summary]: #summary
+
+Improve the management of openSUSE MicroOS in Uyuni.
+
+# Motivation
+[motivation]: #motivation
+
+MicroOS, or the downstream SLE Micro, is a different operating system than e.g. SLES. It is based on the same tools and packages, with few additions on top. These additions are few, but completely change the operating model.
+
+The main idea is: **All changes go into a new btrfs snapshot. This new snapshot is "pending" until a reboot activates it.**
+
+Until now, we tried to hide the differences in Uyuni and relied on Salt to do the *right thing*. This strategy was easy for us to use, but it did not work well. We need to change the approach we take with transactional systems.
+
+# Detailed design
+[design]: #detailed-design
+
+## Uyuni differentiates between OS-unchanging and OS-altering operations
+Instead of using the `transactional_update` executor, Uyuni calls either `state.apply` or `transactional_update.apply`. Uyuni is in full control of which states are applied in a new snapshot and which states are not. The SLS file is the smallest unit Salt can handle, there is no way to apply only parts of an SLS file inside a snapshot. Therefore, we split SLS files that currently mix OS-unchanging and OS-altering operations.
+
+The operations below map to SLS files. The syntax used for operations in this document is the same that's used in for Salt `top.sls`.
+
+### Two approaches to run operations
+
+- `state.apply` and `transactional_update.apply`
+- Java code supports both
+- UI shows both on pages were users can schedule actions
+- API offers a way to choose between the two
+
+### Uncategorized operations
+
+- `virt.*`: About to be dropped
+- `bootloader`: TODO
+- `rebootifneeded` - The way this is written is incompatible with transactional systems
+
+### OS-unchanging operations
+
+These operations do not alter the system, i.e. they don't belong in a (new) snapshot. Uyuni applies them with `state.apply` for two reasons: structured output and no concurrency (`queue=True`).
+
+- `ansible.runplaybook`
```

Comment:
Is `ansible.runplaybook` intended for SLM/MicroOS to act as a control node that executes the playbook against other minions? Because the playbook can be executed against the control node itself as well, which would then be OS-altering of course. 

Then again, if we make this a transactional action, that means the playbook files must be present on the transactional filesystem. We should probably make a note somewhere in the doc at least that we don't support Uyuni/SUMA server to alter itself with Ansible.